crates/sqlexec/src/planner/extension.rs

```diff
@@ -12,8 +12,13 @@ pub trait ExtensionType: Sized + UserDefinedLogicalNodeCore {
             node: Arc::new(self),
         }
     }
+    fn into_logical_plan(self) -> LogicalPlan {
+        LogicalPlan::Datafusion(datafusion::logical_expr::LogicalPlan::Extension(
+            self.into_extension(),
+        ))
+    }
     fn try_decode_extension(extension: &LogicalPlanExtension) -> Result<Self>;
-    fn try_encode(&self, buf: &mut Vec<u8>, codec: &dyn LogicalExtensionCodec) -> Result<()>;
+    fn try_encode(&self, buf: &mut Vec<u8>, _codec: &dyn LogicalExtensionCodec) -> Result<()>;
```

Comment:
Why underscore (since this a trait method)?

Reply:
using `_codec` because many of the implementers don't need the codec. the codec is only needed when you have non leaf nodes. 

So it's there if you need it, but many implementers don't need it. so when you implement the trait, you dont have to manually add the `_` on every implementation that doesn't need it. seemed like a sensible default to me.